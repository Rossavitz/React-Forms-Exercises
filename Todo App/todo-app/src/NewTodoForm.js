import React, { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = {
    task: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData });
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="Add-Todo">Add Todo: </label>
      <input
        id="Add-Todo"
        type="text"
        name="task"
        placeholder="Todo"
        value={formData.task}
        onChange={handleChange}
      />
      <button>Add Todo</button>
    </form>
  );
};

export default NewTodoForm;
