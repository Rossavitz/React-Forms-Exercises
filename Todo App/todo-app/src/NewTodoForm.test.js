import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// smoke test
it("renders without crashing", function () {
  render(<NewTodoForm />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("runs addTodo on form submit", function () {
  const createMock = jest.fn();
  render(<NewTodoForm addTodo={createMock} />);
  const createButton = screen.getByText("Add Todo");
  fireEvent.click(createButton);
  expect(createMock).toHaveBeenCalled();
});
