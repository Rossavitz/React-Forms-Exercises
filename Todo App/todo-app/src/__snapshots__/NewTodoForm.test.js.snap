// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="Add-Todo"
    >
      Add Todo: 
    </label>
    <input
      id="Add-Todo"
      name="task"
      placeholder="Todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
